python/ray/_private/worker.py

```diff
@@ -1232,6 +1232,12 @@ def init(
         logger.debug("Could not import resource module (on Windows)")
         pass
 
+    if job_config:
+        warnings.warn(
+            "DeprecationWarning: job_config is deprecated and will be removed. "
+            "Use runtime_env instead."
```

Comment:
@ericl But I think some config items have not been supported in runtime env yet, right? like default_actor_lifetime, ray_namespace.

Or could you tell how  users set ray_namespace in runtime env after this PR?

Reply:
Putting namespace at the top level looks ok. 
@jianoaix  It's still need to be declared (even supported in this PR) the alternative APIs for the fields in JobConfig.

Reply:
In this PR, we recommend users to use runtime env instead, like:
```python
ray.init(runtime_env={"default_actor_lifetime": "detached"})  # recommended API
```
But we should make sure the above API is ready for using, right?

Reply:
Yes, agree. So do you think this is not ready (then we should drop JobConfig deprecation here), or you think we just need to add this recommendation in the warning message (then we will still deprecate it, and just improve the log)?

Reply:
I believe it's not ready. I think we should support these APIs firstly:
```python
ray.init(runtime_env={"default_actor_lifetime": "detached"})
```

Reply:
Make sense, going to revert job_config.